megamek/src/megamek/common/loaders/BLKBattleArmorFile.java

```diff
@@ -237,13 +231,24 @@ protected void loadEquipment(Entity t, String sName, int nLoc) throws EntityLoad
                     try {
                         Mounted m = t.addEquipment(etype, nLoc, false, 
                                 mountLoc, dwpMounted);
-                        if (numShots != 0 && m != null 
```

Comment:
Entity#addEquipment never returns null. It can throw an exception, but it is guaranteed to be non-null in the try block. The test was confusing code analysis tools into flagging subsequent dereferences as potential NPEs.